hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/updatetokencustomfees/UpdateTokenCustomFeesDecoder.java

```diff
@@ -0,0 +1,186 @@
+                        decodeRoyaltyFees(call.get(ROYALTY_FEE), addressIdConverter))
+                .toList();
+    }
+
+    private Stream<CustomFee> decodeFixedFees(
+            @NonNull final Tuple[] fixedFee,
+            @NonNull final AddressIdConverter addressIdConverter,
+            @NonNull final TokenID tokenId) {
+        return Arrays.stream(fixedFee).map(fee -> CustomFee.newBuilder()
+                .fixedFee(FixedFee.newBuilder()
+                        .amount(fee.get(FIXED_FEE_AMOUNT))
+                        .denominatingTokenId(determineDenominatingToken(fee, tokenId))
+                        .build())
+                .feeCollectorAccountId(addressIdConverter.convert(fee.get(FIXED_FEE_FEE_COLLECTOR)))
+                .build());
+    }
+
+    private Stream<CustomFee> decodeFractionalFees(
+            @NonNull final Tuple[] fractionalFees, @NonNull final AddressIdConverter addressIdConverter) {
+        return Arrays.stream(fractionalFees).map(fee -> CustomFee.newBuilder()
+                .fractionalFee(FractionalFee.newBuilder()
+                        .fractionalAmount(Fraction.newBuilder()
+                                .numerator(fee.get(FRACTIONAL_FEE_NUMERATOR))
+                                .denominator(fee.get(FRACTIONAL_FEE_DENOMINATOR))
+                                .build())
+                        .minimumAmount(fee.get(FRACTIONAL_FEE_MIN_AMOUNT))
+                        .maximumAmount(fee.get(FRACTIONAL_FEE_MAX_AMOUNT))
+                        .netOfTransfers(fee.get(FRACTIONAL_FEE_NET_OF_TRANSFERS))
+                        .build())
+                .feeCollectorAccountId(addressIdConverter.convert(fee.get(FRACTIONAL_FEE_FEE_COLLECTOR)))
+                .build());
+    }
+
+    private Stream<CustomFee> decodeRoyaltyFees(
+            @NonNull final Tuple[] royaltyFees, @NonNull final AddressIdConverter addressIdConverter) {
+        return Arrays.stream(royaltyFees).map(fee -> CustomFee.newBuilder()
+                .royaltyFee(RoyaltyFee.newBuilder()
+                        .exchangeValueFraction(Fraction.newBuilder()
+                                .numerator(fee.get(ROYALTY_FEE_NUMERATOR))
+                                .denominator(fee.get(ROYALTY_FEE_DENOMINATOR))
+                                .build())
+                        .fallbackFee(getFallbackFee(fee))
+                        .build())
+                .feeCollectorAccountId(addressIdConverter.convert(fee.get(ROYALTY_FEE_FEE_COLLECTOR)))
+                .build());
+    }
+
+    // In the solidity structure for Fixed Fees we have the property `bool useCurrentTokenForPayment` that is not
+    // present in the protobuf version.
+    // In order to determine the denominating token we need to check if the property is present.
+    private TokenID determineDenominatingToken(@NonNull final Tuple fee, @NonNull final TokenID tokenId) {
```

Comment:
Here and at L171 for `getIfPresent` and L176 for `getFallbackFee` an explicit `@Nullable` on the return type?

Reply:
Annotations added.